src/player.cpp

```diff
@@ -274,18 +329,27 @@ unsigned int Player::play(
     bool looping,
     double loopingStartAt)
 {
-    auto const &s = std::find_if(sounds.begin(), sounds.end(),
-                                 [&](std::unique_ptr<ActiveSound> const &f)
-                                 { return f->soundHash == soundHash; });
+    // printf("*** PLAYER:PLAY() sounds length: %d  looking for hash: %u\n", sounds.size(), soundHash);
+
+    // for (int i = 0; i < sounds.size(); i++)
+    //     printf("*** PLAYER:PLAY()1 sounds hash: %u\n", sounds[i].get()->soundHash);
+
+    auto const &s = std::find_if(
```

Comment:
I have no idea what I'm saying since I'm not a C++ developer at all, but I wonder — is this finding an object in a list/queue using a hash? Could this be a hashmap instead?

Reply:
(Obviously not relevant to this PR. Feel free to ignore.)

Reply:
Sorry, I'm not making myself clear at all. 

I just meant that if using something like `std::unordered_map` would make the performance better (O(1) access). When it's a list (`std::vector`), like currently, then we need to iterate the whole list every time we try to find something, which makes every such access O(n), if I'm not mistaken.

Reply:
Good catch! Maybe I am wrong but the `find_if` stops at the first occurrence. I write this down for a future PR, I need to dig this.

Reply:
Yes, definitely something to explore in a separate PR. And clearly not a priority because there's seldom a large number of sound in the list anyway.

> Maybe I am wrong but the find_if stops at the first occurrence.

That's what I think, too. But that still means you need to check a number of cases, and that number grows linearly with the number of sounds. Plus it's a `->` check for each sound. If there was a map instead of a list, it's just "give me the sound with this hash" and you get it in constant time.